src/closchema/core.clj

```diff
@@ -174,15 +174,20 @@
 
 
 (defn- find-matching-properties [instance [pattern schema]]
-  (when-let [matches (some #(re-find (re-pattern (name pattern)) (name %)) (keys instance))]
-    (let [prop-name (if (vector? matches) (first matches) matches)]
-      [(keyword prop-name) schema])))
+  (when-let [matches (keep identity
+                           (map #(re-matches (re-pattern (name pattern))
+                                             (name %))
+                                (keys instance)))]
+    (map #(vector (keyword (if (coll? %1) (first %1) %1)) %2)
+         matches (repeat schema))))
 
 
 (defn- absorb-properties
   [pattern-properties instance]
   (if (map? instance)
-    (into {} (remove nil? (map (partial find-matching-properties instance) pattern-properties)))
+    (into {} (reduce concat (remove nil? (map (partial find-matching-properties
+                                                       instance)
+                                              pattern-properties))))
```

Comment:
Since `(reduce concat (map ...))` is just `(mapcat ...)`
and `(concat x nil) => x` i think you can write this simply as:

```clojure
(into {} (mapcat (partial find-matching-properties instance) pattern-properties))
```